cv32e40s/docs/VerifPlans/Simulation/xsecure/xsecure_vplan.csv

```diff
@@ -74,40 +74,45 @@ a_xsecure_hint_updates_lfsr2.",,,,,,,,,,,,,,,,,
-a_glitch_xsecure_hardened_csr_mismatch_pmpaddr,
-a_glitch_xsecure_hardened_csr_mismatch_mtvt,
-a_glitch_xsecure_hardened_csr_mismatch_mtvec,
-a_glitch_xsecure_hardened_csr_mismatch_mintstatus,
-a_glitch_xsecure_hardened_csr_mismatch_mintthresh,
-a_glitch_xsecure_hardened_csr_mismatch_mtvec,
-a_glitch_xsecure_hardened_csr_mismatch_mie",,,,,,,,,,,,,,,,,
+,Hardened CSR,Shadow registers to ensure integrity of important CSR's,If a CSR is not equal to the bit-wise complement of its shadow CSR an alert shall be raised on alert_major_o. Hardened CSR's are listed In the User Manu,"Assert that CSRs and their shadows are the invert of eachother at all times (when not in reset), or they are not alert major is set.",N/A,N/A,N/A,N/A,,,,,,,,,,,,,,,,,
+,,cpuctrl,(See above.),(See above.),Assertion check,Constrained-Random and Directed Self-Checking,Assertion Coverage,"A: a_xsecure_hardened_csr_cpuctrl
+A: a_glitch_xsecure_hardened_csr_mismatch_cpuctrl",,,,,,,,,,,,,,,,,
+,,dcsr,(See above.),(See above.),Assertion check,Constrained-Random and Directed Self-Checking,Assertion Coverage,"A: a_xsecure_hardened_csr_dcsr
+A: a_glitch_xsecure_hardened_csr_mismatch_dcsr",,,,,,,,,,,,,,,,,
+,,jvt,(See above.),(See above.),Assertion check,Constrained-Random and Directed Self-Checking,Assertion Coverage,"A: a_xsecure_hardened_csr_jvt
+A: a_glitch_xsecure_hardened_csr_mismatch_jvt",,,,,,,,,,,,,,,,,
+,,mcause,(See above.),(See above.),Assertion check,Constrained-Random and Directed Self-Checking,Assertion Coverage,"A: 
+A: ",TODO,,,,,,,,,,,,,,,,
+,,mepc,(See above.),(See above.),Assertion check,Constrained-Random and Directed Self-Checking,Assertion Coverage,"A: a_xsecure_hardened_csr_mepc
+A: a_glitch_xsecure_hardened_csr_mismatch_mepc",,,,,,,,,,,,,,,,,
+,,mie,(See above.),(See above.),Assertion check,Constrained-Random and Directed Self-Checking,Assertion Coverage,"A: a_xsecure_hardened_csr_mie
+A: a_glitch_xsecure_hardened_csr_mismatch_mie",,,,,,,,,,,,,,,,,
+,,mintstatus,(See above.),(See above.),Assertion check,Constrained-Random and Directed Self-Checking,Assertion Coverage,"A: a_xsecure_hardened_csr_mintstatus
+A: a_glitch_xsecure_hardened_csr_mismatch_mintstatus",,,,,,,,,,,,,,,,,
+,,mintthresh,(See above.),(See above.),Assertion check,Constrained-Random and Directed Self-Checking,Assertion Coverage,"A: a_xsecure_hardened_csr_mintthresh
+A: a_glitch_xsecure_hardened_csr_mismatch_mintthresh",,,,,,,,,,,,,,,,,
+,,mscratch,(See above.),(See above.),Assertion check,Constrained-Random and Directed Self-Checking,Assertion Coverage,"A: a_xsecure_hardened_csr_mscratch
+A: a_glitch_xsecure_hardened_csr_mismatch_mscratch",,,,,,,,,,,,,,,,,
+,,mscratchcsw,(See above.),(See above.),Assertion check,Constrained-Random and Directed Self-Checking,Assertion Coverage,"A: 
+A: ",TODO,,,,,,,,,,,,,,,,
+,,mscratchcswl,(See above.),(See above.),Assertion check,Constrained-Random and Directed Self-Checking,Assertion Coverage,"A: 
+A: ",TODO,,,,,,,,,,,,,,,,
```

Comment:
I remembered I removed these two CSR checks for a reason, but dont remember the reason. Will update you on this.

Reply:
The mscratch* CSRs share the actual flops from the regular mscratch CSR and that one has shadow flops, so they all have.

Reply:
You can write A: a_xsecure_hardened_csr_mscratch
A: a_glitch_xsecure_hardened_csr_mismatch_mscratch for the mscratchsw/l, and remove the TODOs. I originally removed those items form the vplan, but I think it is most correct to include the vplan points and reuse the assertions.

Reply:
Should be fixed now.

Reply:
Remember to click "Resolve conversation" if all items in discussion is resolved

Reply:
Done.